Makefile

```diff
@@ -42,10 +42,13 @@ FUSES      = -U hfuse:w:0xd8:m -U lfuse:w:0xff:m
 # Tune the lines below only if you know what you are doing:
 
 AVRDUDE = avrdude  $(PROGRAMMER) -p $(DEVICE) -B 10 -F -D
-COMPILE = avr-gcc -Wall -Os -DF_CPU=$(CLOCK) -mmcu=$(DEVICE) -I. -ffunction-sections
+CFLAGS = -Wall -Os -DF_CPU=$(CLOCK) -mmcu=$(DEVICE) -I
+COMPILE = avr-gcc $(CFLAGS) -ffunction-sections
```

Comment:
I am not good at makefiles, but it kinda looks like `COMPILE` gets defined before we add the `-g` to `CLFAGS` down on line 51 when making the debug target. Have you compiled the debug target and gotten the debugging information included in the hexfile?